### PR TITLE
ISSUE #2943 - Upload containers - Destination input box

### LINKED ISSUE #2943: Upload containers - Destination input box
### Description
The select box for creating a new container/select a container when uploading a new model has lots of states . It should consists of two components. An input  and a  select component linked up with an autocomplete component.

### Goals
- [ ] Provide the option to select an already existing container
- [ ] Provide the option to select a new container 
- [ ] Provide a way to show errors 
- [ ] Autocomplete the option

### Tasks
- [ ] make an unstyled version of the component to test that everything can work as intended using the autocomplet
- [ ] Create an input for an existing container following the styles defined in the mockup 
- [ ] Create ( or reuse, because it seems that it was already defined in the UI kit)  an input for new containers
- [ ] Add a tooltip for errors
- [ ] Create a select item for rendering the existing container that will populate the select box (bear in mind the styling)
- [ ] Update the select item for rendering the existing container that will populate the select box  with a disabled style
- [ ] Create a select item for new containers following the styling 

### Related Resources
[MockUp
](https://www.figma.com/file/djgxvRKOaMMyutswn5yywU/Teamspace-(live)?node-id=2874%3A69529)Coda doc: `link to coda`

### PR DESCRIPTION
**Pull request by The-Daniel**
This fixes #2943

#### Description
- Adds autocomplete box for selecting the destination container of an upload
- Containers are disabled if already in use by another upload
- Gives the ability to choose to make a new container if the name is not already in use
- If the box fails validation the error is displayed in a tooltip

#### Test cases
- Add several files and set their containers
- Click the 'x' icon to clear the input. This will also display an error because the container name is empty.
- When existing containers are selected most fields in the sidebar are disabled
- When chosing to add a new container the sidebar fields can be typed
- It is not possible to use the same container twice

### CODE REVIEW

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.styles.ts [hunk 1]
```diff
@@ -16,20 +16,44 @@
...
+export const TextInput = styled(TextField)`
+	margin: 0;
+	width: 271px;
+	border: none;
+	>.MuiInputBase-root {
+		>.MuiInputBase-input {
+			font-weight: bold;
+		}
+		${({ state, theme, error }) => {
+		if (error) return '';
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.styles.ts:32]: I dont think is necessary to handle 'error'. It already returns '' if its not 'new' or 'existing'

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadFileForm.component.tsx [hunk 2]
```diff
@@ -132,7 +132,10 @@ export const UploadFileForm = ({ openState, onClickClose }: IUploadFileForm): JS
 										<SidebarForm
 											value={getValues(`uploads.${selectedIndex}`)}
 											key={fields[selectedIndex].uploadId}
-											isNewContainer={!fields[selectedIndex].containerId}
+											isNewContainer={
+												!getValues(`uploads.${selectedIndex}.containerId`)
+												&& !!getValues(`uploads.${selectedIndex}.containerName`)
```
[the remark this answers is not in the document]
Comment (The-Daniel) [reply frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadFileForm.component.tsx:137]: This is for when a container hasn't been selected yet (i.e. when you first add the file). If no container is set then the inputs need to be disabled. You only know it's a new container when it has a name but doesn't have a containerId

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.styles.ts (hunk 1 above)
Comment (The-Daniel) [reply frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.styles.ts:32]: The problem here was that an input can have a state and an error at the same time (the state just says whether it's a new container, existing container, or is empty) and the state overwrites the error style. I suppose the term 'state' is misleading and I should instead use something else

Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.styles.ts:31]: I understand that, but I tried it and I didnt noticed anything different between one version taking error as a parameter and the other without it.

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.component.tsx [hunk 3]
```diff
@@ -15,34 +15,123 @@
...
+	}, []);
+
+	return (
+		<MuiAutocomplete
+			popupIcon={<ChevronIcon />}
+			closeIcon={<ClearIcon />}
+			openText=""
+			closeText=""
+			clearText=""
+			handleHomeEndKeys
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.component.tsx:69]: Lets put all of these in the theme, props section. Checkout MuiCheckbox to see how the icons are set.

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/uploadListItem.component.tsx [hunk 4]
```diff
@@ -72,9 +72,17 @@ export const UploadListItem = ({
 					field: { ref, ...extras },
 				}) => (
 					<UploadListItemDestination
-						isSelected={isSelected}
+						// isSelected={isSelected}
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/uploadListItem.component.tsx:75]: Delete this one if its not longer being used

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/uploadListItem.component.tsx (hunk 4 above)
Comment (The-Daniel) [reply frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/uploadListItem.component.tsx:75]: oops haha

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**